### DISCUSSION TITLE
Expand Protected Table

### DISCUSSION THREAD

**Discussion by darrenpaul** [pocketbase/pocketbase#1484] [Q&A] [answered]
Hello

I was wondering if its possible to expand a protected table

So I have Collection A which is created by an authenticated user and contains sensitive information which cannot be public. Collection A is a protected table and only viewable by the creator.
When Collection A is created a Collection B is also created with contains a relation to Collection A. Collection B is not a protected table and is viewable publically.

I would just like to pull relative information from Collection A such as a Date, when I request Collection B.

Currently how I'm doing it is adding the information I want from Collection A into Collection B, but if I update Collection A then Collect B needs to be updated, which be nice to just be able to pull the relevant information with an expand.

Examples of Table
Collection A
id, email, date, firstName, lastName

Collection B
id, relation_to_collection_a, date

Any help, or suggestions on how to do what I'm trying to achieve would be amazing.

Thanks

**Comment by ganigeorgiev** [comment, marked as answer]
There is no direct/easy way to do this at the moment without [using PocketBase as framework](https://pocketbase.io/docs/use-as-framework/) and [registering a custom endpoint](https://pocketbase.io/docs/router/) that will query and return only the fields you want. There are plans in the future to add support for "view" collection type but that will be handled as part of https://github.com/pocketbase/pocketbase/issues/311#issuecomment-1312790397.

Alternatively, depending on your use case, you can avoid the duplicated fields by just removing them from your main collection and leave them only in the secondary view-able collection. This may be a little boilerplat-ish since it will require to submit 2 separate requests  when creating/updating (_one to collection A and another with the public fields to B_). Later to fetch the related collection B with your collection A request, you can use [indirect expand](https://pocketbase.io/docs/expanding-relations/#indirect-expand):

```js
const result = await pb.collection('collectionA').getList(1, 30, { expand: 'collectionB(yourRelFieldToA)' })
```

**Comment by darrenpaul** [comment]
Thanks so much, I'll give that a try and wait for the new features 😄
